core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.iceberg.exceptions.ServiceFailureException;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+
+/**
+ * A set of consumers to handle errors for requests for table entities or for namespace entities,
+ * to throw the correct exception.
+ */
+public class ErrorHandlers {
+
+  private ErrorHandlers() {
+  }
+
+  public static Consumer<ErrorResponse> namespaceErrorHandler() {
+    return baseNamespaceErrorHandler().andThen(defaultErrorHandler());
+  }
+
+  public static Consumer<ErrorResponse> tableErrorHandler() {
+    return baseTableErrorHandler().andThen(defaultErrorHandler());
+  }
+
+  /**
+   * Table level error handlers.
+   * Should be chained wih the {@link #defaultErrorHandler}, which takes care of common cases.
+   */
+  private static Consumer<ErrorResponse> baseTableErrorHandler() {
+    return error -> {
+
+      switch (error.code()) {
+        // HttpStatus.SC_NOT_FOUND
+        case 404:
+          if (NoSuchNamespaceException.class.getSimpleName().equals(error.type())) {
+            throw new NoSuchNamespaceException("Resource not found: %s", error);
+          } else {
+            throw new NoSuchTableException("Resource not found: %s", error);
+          }
+        // HttpStatus.SC_CONFLICT
+        case 409:
+          throw new AlreadyExistsException("The table already exists: %s", error);
+      }
+    };
+  }
+
+  /**
+   * Request error handlers specifically for CRUD ops on namespaces.
+   * Should be chained wih the {@linkplain ErrorHandlers#defaultErrorHandler}, which takes care of common cases.
+   */
+  private static Consumer<ErrorResponse> baseNamespaceErrorHandler() {
+    return error -> {
+      switch (error.code()) {
+        // HttpStatus.SC_NOT_FOUND
+        case 404:
+          throw new NoSuchNamespaceException("Namespace not found: %s", error);
+        // HttpStatus.SC_CONFLICT
+        case 409:
+          throw new AlreadyExistsException("Namespace already exists: %s", error);
+        // HttpStatus.SC_UNPROCESSABLE_ENTITY
+        case 422:
+          throw new RESTException("Unable to process request due to bad property updates");
+      }
+    };
+  }
+
+  /**
+   * Request error handler that handles the common cases that are included with all responses,
+   * such as 400, 401, 403, 500, etc.
+   */
+  public static Consumer<ErrorResponse> defaultErrorHandler() {
+    return errorResponse -> {
+      switch (errorResponse.code()) {
+        // HttpStatus.SC_CLIENT_ERROR:
+        case 400:
+          throw new BadRequestException("Unable to process the request as it is somehow malformed: %s", errorResponse);
+        // HttpStatus.SC_UNAUTHORIZED
+        case 401:
+          throw new NotAuthorizedException("Not Authorized: %s", errorResponse);
+        // HttpStatus.SC_FORBIDDEN
+        case 403:
+          throw new ForbiddenException("Forbidden: %s", errorResponse);
+        // HttpStatus.SC_METHOD_NOT_ALLOWED
+        case 405:
+        // HttpStatus.SC_NOT_ACCEPTABLE
```

Comment:
At a minimum, I think 405 and 406 should break. Then move the `throw` from default to after the switch statement to catch both default and breaks.

Reply:
Changed 405 and 406 to break.